src/Math/CircBuffer.h

```diff
@@ -0,0 +1,112 @@
+#ifndef CIRC_BUFFER_H
+#define CIRC_BUFFER_H
+
```

Comment:
Add a file level comment explaining generically how to use it and what overrides are there (like bracket notation accessing)